sdk/python/tests/integration/feature_repos/universal/online_store/singlestore.py

```diff
@@ -0,0 +1,43 @@
+import subprocess
+import time
+from typing import Dict
+
+from testcontainers.core.container import DockerContainer
+
+from tests.integration.feature_repos.universal.online_store_creator import (
+    OnlineStoreCreator,
+)
+
+
+class SingleStoreOnlineStoreCreator(OnlineStoreCreator):
+    def __init__(self, project_name: str, **kwargs):
+        super().__init__(project_name)
+        self.container = (
+            DockerContainer("ghcr.io/singlestore-labs/singlestoredb-dev:latest")
+            .with_exposed_ports(3306)
+            .with_env("USER", "root")
+            .with_env("ROOT_PASSWORD", "test")
+            # this license key is authorized solely for use in SingleStore Feast tests and is subject to strict usage restrictions
+            # if you want a free SingleStore license for your own use please visit https://www.singlestore.com/cloud-trial/
+            .with_env(
+                "LICENSE_KEY",
+                "BGIxODZiYTg1YWUxYjRlODRhYzRjMGFmYTA1OTkxYzgyAAAAAAAAAAABAAAAAAAAACgwNQIZANx4NIXJ7CWvKYYb3wIyRXxBY7fdAnLeSwIYLy2Q0jA124GAkl04yuGrD59Zpv85DVYXAA==",
+            )
```

Comment:
Can you clarify why a license is needed here for a dev image? Does an image make some sort of call home checks? Will it work w/o internet access?

We can probably hide the key in a GH actions variable if necessary, but the tests won't be easily runnable on developer machines.

Reply:
the db won't be able to spin up w/o the license

it will work w/o internet access if you download the dev image beforehand